File: Programmers/binary_search1.py
def binary_search(times, n, start, end):
    while start < end:
        mid = (start + end) // 2
        total = sum(mid // time for time in times)
        if total >= n:
            end = mid
        else:
            start = mid + 1
    return start


def solution(n, times):
    answer = binary_search(times, n, 1, max(times) * n)

    return answer

n = 6
times = [7, 10]
# answer = 28
print(solution(n, times))

File: Programmers/binarySearch1.js
const binarySearch = (times, n, start, end) => {
    let mid, total;
    while(start < end) {
        mid = parseInt((start + end) / 2);
        total = times.reduce((s, time) => s + parseInt(mid / time), 0);
        if(total >= n) {
            end = mid;
        } else {
            start = mid + 1;
        }
    }
    return start;
};

function solution(n, times) {
    var answer = binarySearch(times, n, 1, Math.max.apply(null, times) * n);
    return answer;
}